### DISCUSSION TITLE
NanoPC-T6 missing in script "Make your own distribution"

### DISCUSSION THREAD

**Discussion by quasistatic-setup** [MichaIng/DietPi#6436] [Image Request]
In the current script the NanoPC-T6 is not included.
`bash -c "$(curl -sSfL 'https://raw.githubusercontent.com/MichaIng/DietPi/master/.build/images/dietpi-installer')"`
https://dietpi.com/docs/hardware/#make-your-own-distribution

So I tried R6S/R6C.

![image](https://github.com/MichaIng/DietPi/assets/76494705/d582490a-f3e9-4177-91aa-10002bf19223)

But this failed.
![image](https://github.com/MichaIng/DietPi/assets/76494705/dc967364-2207-4f8f-b443-f875e50ac8e9)

I had to go with generic device.
This way I loose the GPU acceleration.

Is it possible to add the T6 to the list?

**Comment by quasistatic-setup** [comment]
will it be added with the july update?

**Comment by MichaIng** [comment]
Sorry for the late reply. Support for an SBC in `dietpi-installer` and general DietPi support + images are one and the same thing. The existing request will enable both: #6400

Until then, selecting "Generic Device" is probably the best you can do.

GPU acceleration should not be affected by the device choice: Selecting "R6S/R6C" installs the exact same kernel on top of the one that is already installed on FriendlyELEC's images, just as an APT package. Selecting "Generic Device" won't touch the kernel at all, so as a result the same kernel effectively remains. The problem is that the Mesa GPU drivers shipped by Debian (also Bookworm), and libraries GUI software was compiled against, are too old to fully support the RK3588 GPU. For Ubuntu, there is a PPA with very recent drivers and software, but it is not as easy to install those on Debian. Not sure whether the individual packages can be installed or whether they cause DEB package conflicts:
- https://launchpad.net/~liujianfeng1994/+archive/ubuntu/panfork-mesa
- https://launchpad.net/~liujianfeng1994/+archive/ubuntu/rockchip-multimedia

I'm closing this in favour of #6400.
